esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -271,6 +270,101 @@ EspHttpBinding::EspHttpBinding(IPropertyTree* tree, const char *bindname, const
+}
+
+void EspHttpBinding::setSDSSession()
+{
+    VStringBuffer xPath("%s/%s[@name=\"%s\"]/%s[@name=\"%s\"]", PathSessionRoot, PathSessionProcess,
+        processName.get(), PathSessionDomain, domainName.str());
+    VStringBuffer appStr("%s[@port=\"%d\"]", PathSessionApplication, m_port);
+    Owned<IRemoteConnection> conn = querySDS().connect(xPath.str(), myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+    if (!conn)
+        throw MakeStringException(-1, "Failed to connect SDS DomainSession.");
+
+    IPropertyTree* sessionDomain = conn->queryRoot();
+    IPropertyTree* appSessionTree = sessionDomain->queryBranch(appStr.str());
+    if (!appSessionTree)
+    {
+        IPropertyTree* newAppSessionTree = sessionDomain->addPropTree(PathSessionApplication, createPTree());
+        newAppSessionTree->addPropInt("@port", m_port);
+    }
+    domainSessionSDSPath.setf("%s/%s/", xPath.str(), appStr.str());
+    sessionIDCookieName.setf("%s%d", SESSION_ID_COOKIE, m_port);
+}
+
+void EspHttpBinding::readAuthDomainCfg(IPropertyTree* procCfg)
+{
+    domainAuthType = AuthPerRequestOnly;
+    VStringBuffer xpath("AuthDomains/AuthDomain[@domainName=\"%s\"]", domainName.get());
+    IPropertyTree* authDomainTree = procCfg->queryPropTree(xpath);
+    if (authDomainTree)
+    {
+        const char* authType = authDomainTree->queryProp("@authType");
+        if (isEmptyString(authType) || strieq(authType, "AuthTypeMixed"))
+            domainAuthType = AuthTypeMixed;
+        else if (strieq(authType, "AuthPerSessionOnly"))
+            domainAuthType = AuthPerSessionOnly;
+        if (domainAuthType == AuthPerRequestOnly)
+            return;
+
+        unsigned sessionTimeoutMinutes = authDomainTree->getPropInt("@sessionTimeoutMinutes", 0);
+        sessionTimeoutSeconds = sessionTimeoutMinutes < 0 ? -1 : sessionTimeoutMinutes > 0 ? sessionTimeoutMinutes * 60 : ESP_SESSION_TIMEOUT;
+
+        //The @unrestrictedResources contains URLs which may be used before a user is authenticated.
+        //For example, an icon file on the login page.
+        const char* unrestrictedResources = authDomainTree->queryProp("@unrestrictedResources");
+        if (!isEmptyString(unrestrictedResources))
+        {
+            StringArray urlArray;
+            urlArray.appendListUniq(unrestrictedResources, ",");
+            ForEachItemIn(i, urlArray)
+            {
+                const char* url = urlArray.item(i);
+                if (isEmptyString(url))
+                    continue;
+                if (isWildString(url))
+                    domainAuthResourcesWildMatch.append(url);
+                else
+                    domainAuthResources.setValue(url, true);
+            }
+        }
+
+        const char* _loginURL = authDomainTree->queryProp("@logonURL");
+        if (!isEmptyString(_loginURL))
+            loginURL.set(_loginURL);
+        else
+            loginURL.set("/esp/files/eclwatch/templates/Login.html");
```

Comment:
I guess I should use a DEFAULT_LOGIN_URL here: static const char* const DEFAULT_LOGIN_URL = "/esp/files/eclwatch/templates/Login.html";